clustering/kubernetes.py

```diff
@@ -0,0 +1,436 @@
+#!/usr/bin/python
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>
+
+DOCUMENTATION = '''
+---
+module: kubernetes
+version_added: "2.0"
+short_description: Manage Kubernetes resources.
+description:
+    - This module can manage Kubernetes resources on an existing cluster using
+      the Kubernetes server API. Users can specify in-line API data, or
+      specify an existing Kubernetes YAML file. Currently, this module,
+        Only supports HTTP Basic Auth
+        Only supports 'strategic merge' for update, http://goo.gl/fCPYxT
+        SSL certs are not working, use 'validate_certs=off' to disable
+      This module can mimic the 'kubectl' Kubernetes client for commands
+      such as 'get', 'cluster-info', and 'version'. This is useful if you
+      want to fetch full object details for existing Kubernetes resources.
+options:
+  api_endpoint:
+    description:
+      - the IPv4 API endpoint of the Kubernetes cluster
+    required: true
+    default: null
+    aliases: ["endpoint"]
+  inline_data:
+    description:
+      - the Kubernetes YAML data to send to the server endpoint
```

Comment:
The `endpoint` word could be highlighted by `I(endpoint)` to emphasise that it refers to the `api_endpoint` resp. `endpoint` option defined above.